code/_onclick/hud/credits.dm

```diff
@@ -23,39 +23,29 @@ GLOBAL_LIST(end_titles)
 			GLOB.end_titles += "<br>"
 
 		GLOB.end_titles += "<center><h1>Thanks for playing!</h1>"
-
-	LAZYINITLIST(credits)
-	var/list/_credits = credits
-	verbs += /client/proc/ClearCredits
-	_credits += new /obj/screen/credit/title_card(null, null, src, SSticker.mode.title_icon)
+	for(var/client/C in GLOB.clients)
+		if(C.prefs.show_credits)
+			C.screen += new /obj/screen/credit/title_card(null, null, SSticker.mode.title_icon)
 	sleep(CREDIT_SPAWN_SPEED * 3)
-	for(var/I in GLOB.end_titles)
-		if(!credits)
-			return
-		_credits += new /obj/screen/credit(null, I, src)
-		sleep(CREDIT_SPAWN_SPEED)
-	sleep(CREDIT_ROLL_SPEED - CREDIT_SPAWN_SPEED)
-	ClearCredits()
-	verbs -= /client/proc/ClearCredits
-
-/client/proc/ClearCredits()
-	set name = "Hide Credits"
-	set category = "OOC"
-	verbs -= /client/proc/ClearCredits
-	QDEL_LIST(credits)
+	for(var/i in 1 to GLOB.end_titles.len)
+		var/C = GLOB.end_titles[i]
+		if(!C)
+			continue
+		addtimer(CALLBACK(GLOBAL_PROC, /proc/create_credit, C), (CREDIT_SPAWN_SPEED+(i*10)), TIMER_CLIENT_TIME)

```

Comment:
Shouldn't this just be `CREDIT_SPAWN_SPEED * i`

Reply:
Not unless you want all of the text to stack on top of itself: https://i.imgur.com/3cgqcie.png

This is an addtimer, it doesn't pause the for loop, it just runs the code after the time has passed.

Reply:
I'm aware of how it works, but doesn't multiplying it by `i` make it have an increasing time before each one spawns?

Reply:
I'm not sure you did what I said right, could you show me the line of code you put in?

Reply:
Use discord damn it, this is way more tedious.